content/en/agent/faq/linux-key-rotation-2024.md

```diff
@@ -0,0 +1,201 @@
+---
+title: 2024 Linux Key Rotation
+kind: guide
+private: true
+aliases:
+  - /agent/guide/linux-key-rotation-2024
+---
+
+As a common best practice, Datadog periodically rotates the keys and certificates used to sign Datadog's Agent packages. The Agent packages includes:
+
+- the different flavours of agent (datadog-agent, datadog-iot-agent, datadog-heroku-agent, datadog-dogstatsd, datadog-fips-proxy and datadog-agent-dbg).
+- additional agent-related packages: observability pipeline worker and the APM tracer libraries (java, python, .Net, ruby and js).
+
+The following GPG keys, used to sign the above RPM and DEB packages, reach their end-of-life in September 2024. The rotation is planned for June 2024:
+
+- The RPM signing key with hash [`C6559B690CA882F023BDF3F63F4D1729FD4BF915`][1] rotation is planned on June, 2024 (exact date & time to be communicated later) and replaced by the key with hash [`7408BFD56BC5BF0C361AAAE85D88EEA3B01082D3`][2]. The new trusted key needs to be installed prior installing any RPM release published after this date.
+- The DEB signing key with hash [`D75CEA17048B9ACBF186794B32637D44F14F620E`][3] rotation is planned on June, 2024 (exact date & time to be communicated later) and replaced by the key with hash [`5F1E256061D813B125E156E8E6266D4AC0962C7D`][4]. APT checks the repo metadata signature, so the new key needs to be trusted by this date to install any future or existing version of the Agent from `apt.datadoghq.com`.
+
+Customers using Datadog's RPM or DEB packages might require a manual action to import the new key on their systems to install or upgrade the Agent packages after the rotation takes place.
+
+<div class="alert alert-info">
+<strong>Note</strong>: This DOES NOT affect the functionality of already running Agents. It only limits the ability to install or upgrade to a newer version of the Agent. Also, this doesn't affect Dockerized Linux Agents, Windows, or macOS Agents.
+</div>
+
+## Install methods that automatically trust the new GPG key
+
+Your host automatically trusts the new key (no further action is required) if you're using one of the following install methods:
+
+- [Agent install script][5] v1.18.0+ (released Jun 27, 2023)
+- [Chef cookbook][6] v4.18.0+ (released Jul 26, 2023)
+- [Ansible role][7] v4.20.0+ (released Jul 18, 2023)
+- [Ansible collection][14] v5.0.0+ (released Jul 18, 2023)
+- [Puppet module][8] v3.21.0+ (released Jul 05, 2023)
+- [SaltStack formula][9] v3.6+ (released Aug 10, 2023)
+- [Heroku buildpack][10] v2.11+ (released Jun 15, 2023)
+- [Elastic Beanstalk][11] config templates updated as of Nov 11, 2022 or later (should contain `source: https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh`)
```

Comment:
We released the install_script on June 27 2023, so the beanstalk date should probably be no sooner than that, right?

Reply:
Indeed, I took the date where it mentions install_script_agent7 for the first time...